fvtest/tril/tril/ilgen.cpp

```diff
@@ -184,11 +185,37 @@ TR::Node* Tril::TRLangBuilder::toTRNode(const ASTNode* const tree) {
         }
      }
      else if (opcode.isLoadIndirect() || opcode.isStoreIndirect()) {
-         auto offset = tree->getArgByName("offset")->getValue()->get<int32_t>();
-         TraceIL("  is indirect store/load with offset %d\n", offset);
+         TraceIL("  is indirect store/load with ");
+         // If not specified, offset will default to zero. 
+         int32_t offset = 0; 
+         if (tree->getArgByName("offset")) {
+            offset = tree->getArgByName("offset")->getValue()->get<int32_t>();
+         } else { 
+            TraceIL(" (default) ");
+         }
+         TraceIL("offset %d ", offset);
+
          const auto name = tree->getName();
-         auto type = opcode.getType();
          auto compilation = TR::comp();
+         TR::DataType type;  
+         if (opcode.isVector()) { 
+            // Vector types in TR IL are "typeless", insofar as they are
+            // supposed to infer the vector type depending on the children.
+            // Loads determine their data type based on the symref. However,
+            // given that we are creating a symref here, we need a hint as to
+            // what type of symref to create. So, vloadi and vstorei will take 
+            // an extra argument "type" to annotate the type desired.
+            if (tree->getArgByName("type") != nullptr) { 
```

Comment:
Shouldn't this `nullptr` be `NULL` as required by #1718 ?

Reply:
I had been matching the style of the file as it was, but you're right, this is going to make merging easier. Fixed. Ditto below.